extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts

```diff
@@ -21,12 +22,22 @@ export class BdcDashboardModel {
 	private _endpointsLastUpdated: Date;
 	private readonly _onDidUpdateEndpoints = new vscode.EventEmitter<EndpointModel[]>();
 	private readonly _onDidUpdateBdcStatus = new vscode.EventEmitter<BdcStatusModel>();
+	private readonly _onError = new vscode.EventEmitter<Error>();
 	public onDidUpdateEndpoints = this._onDidUpdateEndpoints.event;
 	public onDidUpdateBdcStatus = this._onDidUpdateBdcStatus.event;
+	public onError = this._onError.event;
 
 	constructor(private options: BdcDashboardOptions, ignoreSslVerification = true) {
-		this._clusterController = new ClusterController(options.url, options.auth, options.username, options.password, ignoreSslVerification);
-		this.refresh();
+		try {
+			this._clusterController = new ClusterController(options.url, options.auth, options.username, options.password, ignoreSslVerification);
+			this.refresh();
+		} catch {
```

Comment:
what error are you trying to catch here, the new ClusterController()? this.refresh is async and the error won't be caught here

Reply:
ClusterController throws an error in the constructor if the username or password is empty and basic auth is selected